mozart/services/ci.py

```diff
@@ -212,11 +236,22 @@ def get(self, job_name=None):
     @build_job_ns.expect(parser)
     def post(self, job_name=None):
         """start Jenkins build"""
-        request_data = request.json or request.form
+        request_data = request.form
 
         if job_name is None:
             repo = request_data.get('repo')
             branch = request_data.get('branch')
+
+            # Check json if repo not found - see JobRegistration.post() above.
+            if repo is None:
```

Comment:
there are 2 `if repo is None`, should we move this block inside the other one to avoid redundancy

Reply:
We need two separate checks for repo is None - the first is to see if it wasn't in the request.form and need to check request.json, the second is to see if both request.form and request.json failed to produce a value. I suppose the second check could be moved into the first but it would have to be outside of the try/except. Thoughts?

Reply:
I see. Could try:
```
repo = request.form.get("repo", request.json.get("repo"))
branch = request.form.get("branch", request.json.get("branch"))
if repo is None:
```

Reply:
Ah, yeah, forgot about the default to get. I like that - I will check that in.

Reply:
Actually, this could generate an unnecessary error. Let's say that branch was not specified (as it is optional) - the second line above would incorrectly generate the 400 error if data is passed in request.form and not request.json.

Reply:
i see. so the current implementation accounts for that as well?